core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -305,7 +305,7 @@ private[spark] class MapOutputTrackerMaster(conf: SparkConf)
 
     if (mapStatuses.contains(shuffleId)) {
       val statuses = mapStatuses(shuffleId)
-      if (statuses.nonEmpty) {
+      if (statuses.nonEmpty && statuses.exists(_ != null)) {
```

Comment:
same thing here -- the additional test case passes w/out this test.  If it fixes another issue, we should have a better test case.